big_tests/tests/mod_event_pusher_rabbit_SUITE.erl

```diff
@@ -0,0 +1,321 @@
+
+connected_users_push_presence_events_when_change_status(Config) ->
+    escalus:story(
+      Config, [{bob, 1}, {alice, 1}],
+      fun(Bob, Alice) ->
+              %% GIVEN
+              escalus:wait_for_stanzas(Bob, 1),
+              escalus:wait_for_stanzas(Alice, 1),
+              BobJID = nick_to_jid(bob, Config),
+              AliceJID = nick_to_jid(alice, Config),
+              BobRoutingKey = presence_rk(BobJID),
+              AliceRoutingKey = presence_rk(AliceJID),
+              listen_to_presence_events_from_rabbit([BobJID, AliceJID], Config),
+              %% WHEN users generate some traffic.
+              send_presence_stanzas([Bob, Alice], 10),
+              %% THEN  wait for presence events from Rabbit.
+              ?assertReceivedMatch({#'basic.deliver'{
+                                       routing_key = BobRoutingKey},
+                                    #amqp_msg{}}, timer:seconds(5)),
+              ?assertReceivedMatch({#'basic.deliver'{
+                                       routing_key = AliceRoutingKey},
+                                    #amqp_msg{}}, timer:seconds(5))
+      end).
+
+connected_users_create_exchanges_when_available(Config) ->
+    escalus:story(
+      Config, [{bob, 1}, {alice, 1}],
+      fun(Bob, Alice) ->
+              %% GIVEN
+              Connection = proplists:get_value(rabbit_connection, Config),
+              Channel = proplists:get_value(rabbit_channel, Config),
+              BobJID = nick_to_jid(bob, Config),
+              AliceJID = nick_to_jid(alice, Config),
+              Exchanges = [BobJID, AliceJID],
+              %% WHEN users login
+              escalus:wait_for_stanzas(Bob, 1),
+              escalus:wait_for_stanzas(Alice, 1),
+              %% THEN exchanges are created
+              Result = [is_exchange_present(Connection, Channel, Exchange,
+                                 ?IF_EXCHANGE_EXISTS_RETRIES)
+                        || Exchange <- Exchanges],
+              ?assert(lists:all(fun(E) -> E == true end, Result))
+      end).
+
+disconnected_users_delete_exchanges_when_unavailable(Config) ->
+    escalus:story(
+      Config, [{bob, 1}, {alice, 1}],
+      fun(Bob, Alice) ->
+              %% GIVEN
+              Connection = proplists:get_value(rabbit_connection, Config),
+              Channel = proplists:get_value(rabbit_channel, Config),
+              BobJID = nick_to_jid(bob, Config),
+              AliceJID = nick_to_jid(alice, Config),
+              Exchanges = [BobJID, AliceJID],
+              %% WHEN users login and logout
+              escalus:wait_for_stanzas(Bob, 1),
+              escalus:wait_for_stanzas(Alice, 1),
+              escalus:send(Bob, escalus_stanza:presence(<<"unavailable">>)),
+              escalus:send(Alice, escalus_stanza:presence(<<"unavailable">>)),
+              %% THEN exchanges are deleted
+              [?assertException(_Error, _Reason,
```

Comment:
I find it unusual to write helpers, which raises an Exception and we assert on Exception it raised. I think `error_when_exchange_is_absent/4` should be changed, see my comments on it.